frontend/src/app/commonComponents/Pagination.tsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import classnames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faAngleLeft, faAngleRight } from "@fortawesome/free-solid-svg-icons";
+import "./Pagination.scss";
+import { NavLink } from "react-router-dom";
+
+interface Props {
+  baseRoute: string;
+  totalEntries: number;
+  entriesPerPage: number;
+  currentPage: number;
+  pageGroupSize?: number;
+  className?: string;
+}
+
+// Rules:
+// * Always show first/last page (may be the same page!)
+// * Show current page and N pages on either side
+// * Use elipsis if gaps between group and first/last
+//
+// Cases (_n_ represents currentPage):
+//  << _1_ >>
+//  << 1 _2_ 3 >>
+//  << _1_ 2 3 4 5 ... 71  >>
+//  << 1 2 _3_ 4 5 ... 71  >>
+//  << 1 ... 14 15 _16_ 17 18 ... 71 >>
+//  << 1 ... 67 68 _69_ 70 71 >>
+//  << 1 ... 67 68 69 70 _71_ >>
+//  << 1 2 3 _4_ 5 6 ... 71 >>
+
+// Always make this odd, current page is in the middle
+const defaultGroupSize = 7;
+
+const Pagination = ({
+  baseRoute,
+  currentPage: rawCurrentPage,
+  entriesPerPage,
+  totalEntries,
+  pageGroupSize = defaultGroupSize,
+  className,
+}: Props) => {
+  const groupGutter = Math.floor(pageGroupSize / 2);
+  const totalPages = Math.ceil(totalEntries / entriesPerPage);
+  const currentPage = Math.min(Math.max(+rawCurrentPage || 0, 1), totalPages);
+  const minGroupPage = Math.max(1, currentPage - groupGutter);
+  const maxGroupPage = Math.min(currentPage + groupGutter, totalPages);
+  const pageList = [];
+
+  const Link = (props: {
+    to: string | number;
+    active?: boolean;
+    label?: string;
+    children: React.ReactNode;
+  }) => (
+    <NavLink
```

Comment:
You'll need to use the custom `LinkWithQuery` component or else you will lose the `facility` query parameter and, when you have more than one facility in an organization, you'll need to reselect your facility on each page!

Reply:
This is the impetus for this, btw: https://github.com/CDCgov/prime-simplereport/issues/945

It's _so_ easy to forget to tack on the query param and, even when you do, it's subject to accidental manipulation by users